File: app/src/main/java/ca/judacribz/gainzassist/activities/start_workout/fragments/WarmupsList.java
package ca.judacribz.gainzassist.activities.start_workout.fragments;

import static ca.judacribz.gainzassist.adapters.WorkoutPagerAdapter.EXTRA_WARMUPS;
import static ca.judacribz.gainzassist.models.Exercise.SetsType.WARMUP_SET;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import ca.judacribz.gainzassist.R;
import ca.judacribz.gainzassist.activities.start_workout.StartWorkoutActivity;
import ca.judacribz.gainzassist.models.Exercise;

public class WarmupsList extends Fragment {

    // Constants
    // --------------------------------------------------------------------------------------------
    // --------------------------------------------------------------------------------------------

    // Global Vars
    // --------------------------------------------------------------------------------------------
    StartWorkoutActivity act;
    Bundle bundle;

//    @BindView(R.id.ll_exercise_subtitle_insert) LinearLayout llExSubInsert;
    @BindView(R.id.ll_exercise_sets_insert) LinearLayout llExSetsInsert;
    // --------------------------------------------------------------------------------------------

    // ######################################################################################### //
    // WarmupsList Constructor/Instance                                                        //
    // ######################################################################################### //
    public WarmupsList() {
        // Required empty public constructor
    }

    public static WarmupsList getInstance() {
        return new WarmupsList();
    }

    // ######################################################################################### //

    // Fragment Override
    ///////////////////////////////////////////////////////////////////////////////////////////////
    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        act = (StartWorkoutActivity) context;
        bundle = this.getArguments();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }
View view;
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {
        if (view != null) {
            return  view;
        }
        // Inflate the layout for this fragment
        view =  inflater.inflate(R.layout.fragment_warmups_list, container, false);
        ButterKnife.bind(this, view);

        if (bundle != null) {
            ArrayList<Exercise> warmups = bundle.getParcelableArrayList(EXTRA_WARMUPS);
            if (warmups != null) {

                for (int i = warmups.size()-1; i >= 0; --i) {
                    act.displaySets(
                            WARMUP_SET,
                            warmups.get(i),
                            llExSetsInsert
                    );
                }
            }
        }

        return view;
    }
    //Fragment//Override///////////////////////////////////////////////////////////////////////////
}

File: app/src/main/java/ca/judacribz/gainzassist/activities/start_workout/fragments/WarmupsList.kt
package ca.judacribz.gainzassist.activities.start_workout.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import ca.judacribz.gainzassist.activities.start_workout.StartWorkout
import ca.judacribz.gainzassist.adapters.WorkoutPagerAdapter.Companion.EXTRA_WARMUPS
import ca.judacribz.gainzassist.databinding.FragmentWarmupsListBinding
import ca.judacribz.gainzassist.models.Exercise
import ca.judacribz.gainzassist.models.Exercise.SetsType.WARMUP_SET
import org.parceler.Parcels
import java.util.*

class WarmupsList : Fragment() {

    private var warmups: ArrayList<Exercise>? = null

    private lateinit var binding: FragmentWarmupsListBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentWarmupsListBinding.inflate(inflater, container, false)
        val args = arguments
        if (args != null) {
            warmups = Parcels.unwrap(args.getParcelable(EXTRA_WARMUPS))
        }
        return binding.root
    }

    override fun onResume() {
        super.onResume()
        if (binding.llExerciseSetsInsert.childCount > 0) {
            binding.llExerciseSetsInsert.removeAllViews()
        }
        if (warmups != null) {
            for (warmup in warmups!!) {
                (activity as StartWorkout?)?.displaySets(WARMUP_SET, warmup, binding.llExerciseSetsInsert)
            }
        }
    }

    companion object {
        @JvmStatic
        fun getInstance(): WarmupsList {
            return WarmupsList()
        }
    }
}
